aws/rust-runtime/aws-config/src/lib.rs

```diff
@@ -344,6 +347,73 @@ mod loader {
             self
         }
 
+        /// Provides the ability to programmatically override the profile files that get loaded by the SDK.
+        ///
+        /// The [`Default`] for `ProfileFiles` includes the default SDK config and credential files located in
+        /// `~/.aws/config` and `~/.aws/credentials` respectively.
+        ///
+        /// Any number of config and credential files may be added to the `ProfileFiles` file set, with the
+        /// only requirement being that there is at least one of them. Custom file locations that are added
+        /// will produce errors if they don't exist, while the default config/credentials files paths are
+        /// allowed to not exist even if they're included.
```

Comment:
I feel like this could be reworded a tiny bit. I tried my hand at it, but feel free to ignore this if you like the current wording.

```suggestion
        /// Any number of config and credential files may be added to the `ProfileFiles` file set, with the
        /// only requirement being that there is at least one of each. Profile file locations will produce an
        /// error if they don't exist, but the default config/credentials files paths are exempt from this validation.
```